--- old/lib/compat.h	2017-10-06 07:00:00.000000000 -0400
+++ new/lib/compat.h	2021-03-28 00:19:50.891360075 -0400
@@ -92,9 +92,9 @@
 #if defined(HAVE__OPEN)
 #define open(a, b, c)	_open((a), (b))
 #endif
-#if defined(HAVE__SNPRINTF)
-#define snprintf	_snprintf
-#endif
+//#if defined(HAVE__SNPRINTF)
+//#define snprintf	_snprintf
+//#endif
 #if defined(HAVE__STRDUP)
 #if !defined(HAVE_STRDUP) || defined(_WIN32)
 #undef strdup
